jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -323,12 +348,17 @@ YUI.add('deployment-flow', function() {
       }
       this.setState({deploying: true});
       const args = {
-        credential: this.state.credential,
+        config: {},
         cloud: this.state.cloud && this.state.cloud.name || undefined,
+        credential: this.state.credential,
         region: this.state.region
       };
       if (this.state.sshKey) {
-        args.config = {'authorized-keys': this.state.sshKey};
+        args.config['authorized-keys'] = this.state.sshKey;
+      }
+      if (this.state.vpcId) {
```

Comment:
No I think I really mean `if (this.state.vpcId)`, so that we don't send an empty string.